### ISSUE TITLE
Merge?

### ISSUE THREAD

**Issue by ianks** [willbryant/digest-blake3#6]
Hey there!

I was working on publishing an [initial release of `digest-blake`](https://github.com/Shopify/digest-blake3), when Rubygems told me that this gem already exists 🤦🏻.

Obviously, there's a lot of overlap between our two implementations, but separate approaches. In the one I wrote, we precompile binary gems to help ease the Rust installation problem (rather than bundling the C). We are going to continue using the Rust version due to safety requirements, and I was curious if you would be interested in transferring ownership of the `digest-blake3` name? That way, users could upgrade and get the same behavior, and you would have one less gem to maintain.

If you are interested in this, let me know!

**Comment by willbryant** [comment]
Not really sorry :(. I think the C implementation is more portable than trying to keep on top of pre-building for each platform people might want.

I would be happy to merge a PR to optionally build using the Rust implementation when Rust is available though, would that work?

Also, please note that there is a Rust binding implementation already, https://github.com/Yamaguchi/blake3rb. Perhaps they would like to merge with yours to start with?

**Comment by ianks** [comment]
> Also, please note that there is a Rust binding implementation already, https://github.com/Yamaguchi/blake3rb. Perhaps they would like to merge with yours to start with?

Yeah... that one is old and won't build anymore, plus there would be some breaking API differences. Our versions are a perfect fit since we both went for the `Digest` approach, which I prefer btw!

> I think the C implementation is more portable than trying to keep on top of pre-building for each platform people might want.

Fair enough. I know this may not sway your opinion, but for what it's worth all of that process is entirely automated right now. From my experience, precompiled gems tend to be [more reliable](https://nokogiri.org/tutorials/installing_nokogiri.html#faster-more-reliable-installation) as well.

![image](https://github.com/willbryant/digest-blake3/assets/3303032/3bc1210a-9512-4fc0-a635-14b5f9b5352a)

For truly obscure platforms building from source with Rust isn't a big deal these days (we do it all the time for `ruby-head`, for example).
